Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/VaccinatedPerson.kt

```diff
@@ -79,15 +79,24 @@ data class VaccinatedPerson(
 
     fun getDaysUntilImmunity(nowUTC: Instant = Instant.now()): Int? {
         val newestFullDose = getNewestFullDose() ?: return null
-        val today = nowUTC
-            .toLocalDateUserTz()
-
-        return IMMUNITY_WAITING_DAYS - Days.daysBetween(newestFullDose.vaccinatedOn, today).days
+        val today = nowUTC.toLocalDateUserTz()
+        return if (isPreviousVaccinationAfterRecovery()) 0
+        else IMMUNITY_WAITING_DAYS - Days.daysBetween(newestFullDose.vaccinatedOn, today).days
     }
 
     private fun getNewestDoseVaccinatedOn(): LocalDate =
         vaccinationCertificates.maxOf { it.vaccinatedOn }
 
+    private fun isPreviousVaccinationAfterRecovery(): Boolean {
+        val previousDose = vaccinationCertificates
```

Comment:
What is this method covering?

Reply:
This is for the Vaccination Status info box. It is covering a case where a vaccination 2/2 < 14 days is scanned after a vaccination 1/1 that was done after a revocery. In this case, the new vaccine get highest priority after 14 days, but the info box should state that all planned vaccinations are done and protection is complete. Usually, when you get a dose 2/2 < 14 days, after a dose 1/2,  the info box says vaccination is complete but protection won't be complete for "x" number of days.

Reply:
I've had a talk with the iOS team and @mlenkeit and it looks like this added logic is not actually needed. It was a misinterpretation on my side that we also need to check the previous DCC to see if it's 1/1 and then modify the Vaccination Status Info Box. I will remove it.